### DISCUSSION TITLE
Button in an Option. onBlur causing issues

### DISCUSSION THREAD

**Discussion by dbousamra** [JedWatson/react-select#4862] [Q&A]
Hi all.

I have this example https://codesandbox.io/s/codesandboxer-example-forked-4gkhn?file=/example.tsx

I want to have a button inline that triggers some sort of floating overlay. The issue is that when you click the button, the focus is shifted, and that triggers the menu to close.

How can i implement such a feature?

**Comment by ebonow** [comment]
You seem to be on the right track from looking at your implementation. 

Two other considerations:
1. You may want to introduce an `onBlur` callback prop that checks that if the overlay is toggled off and the menu is still open, then set `menuIsOpen` to undefined instead of false, as false will prevent the menu from opening normally.
2. It seems that you may want to consider adding styling through the styles api to the menuList so it's dimmed and its overflow is set to hidden while the popup is showing to avoid the scrolling menu while popped up.

Let me know if this isn't clear and I can create a forked codesandbox.

**Reply by dbousamra** [reply]
Hey @ebonow 

I ended up solving everything, thanks. The key was to make sure I had:

```
onMouseDown={(e) => {
  e.stopPropagation();
}}
onKeyDown={(e) => {
  e.stopPropagation();
}}
```

on my Option component. This stopped the menu from closing

**Reply by dbousamra** [reply]
If you are interested, here's a video:


https://user-images.githubusercontent.com/648137/137833103-4653665c-947f-4b3f-b2f5-caace6ad7d16.mp4

**Reply by ebonow** [reply]
Fantastic job! It looks really great. 

Your codesandbox had an input in it so I was more concerned about the input than the button to launch it. Glad you have it working.

**Reply by cubecleveland** [reply]
@dbousamra how did u end up clocing the menu ? did u catch a click outside ?
